src/adapters/github.js

```diff
@@ -205,11 +206,81 @@ class GitHub extends Adapter {
   _getTree(path, opts, cb) {
     this._get(`/git/trees/${path}`, opts, (err, res) => {
       if (err) cb(err)
-      else cb(null, res.tree)
+      else {
+        const diff = this._getPatch(path)
+        const diffExists = (Object.keys(diff).length > 0)
+        const filteredTree = res.tree.filter((node) => {
+          node.patch = diff[node.path]
+          delete diff[node.path]
+          return !diffExists || node.patch
+        })
+
+        cb(null, filteredTree)
+      }
     })
   }
 
   // @override
+   _getPatch(path) {
+     const diff = {}
+     const files = $(".diff-view .file-info")
```

Comment:
IMO, the code would be more resilient if we can get patches via API instead of scraping the page. It's my experience of maintaining Octotree that GitHub changes the DOM quite frequently and this code might break soon. If API is not possible, then consider providing a fallback so that Octotree works normally even if the DOM changes, i.e. cancel retrieving the patches and not showing them if the DOM is not as expected.